addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -183,17 +265,22 @@ def format_alphanumeric(text_to_convert):
             or (partner.country_id.code == 'IT' and '0000000')
             or 'XXXXXXX')
 
-        # Represent if the document is a reverse charge refund in a single variable
-        rc_refund = self.move_type == 'in_refund' and document_type in ['TD16', 'TD17', 'TD18']
-
         # Self-invoices are technically -100%/+100% repartitioned
         # but functionally need to be exported as 100%
         document_total = self.amount_total
         if is_self_invoice:
             document_total += sum([abs(v['tax_amount_currency']) for k, v in tax_details['tax_details'].items()])
-            if rc_refund:
+            if reverse_charge_refund:
                 document_total = -abs(document_total)
 
+        conversion_rate = float_repr(
+            self.currency_id._get_conversion_rate(self.currency_id, self.env.ref('base.EUR'), self.company_id, self.invoice_date),
```

Comment:
@dbkosky This one should be calculated based on what is really applied on the invoice, so by dividing balance and amount_currency (or the total fields on the move)